src/endpoint/s3/ops/s3_post_bucket_delete.js

```diff
@@ -23,6 +23,8 @@ async function post_bucket_delete(req) {
     }
 
     dbg.log3('post_bucket_delete: objects', objects);
+    objects = _.uniqWith(objects, _.isEqual);
```

Comment:
I'm not sure about this uniqWith()... 
googled it a bit and saw that it has bad performance, how much time it takes for 1000 objects?
I would use filter() & a helper Set() but we need to check what's faster

Reply:
1. The helper Set() would not preserve the order of the keys.
2. I did a simple check:

<img width="574" alt="Screenshot 2023-06-13 at 8 50 12" src="https://github.com/noobaa/noobaa-core/assets/57721533/717254aa-2a1c-463c-8b41-6a8e8c7a0ed2">

Call delete objects with the same key that appears 1,000 times in the list.
It took less than 0.005 seconds (with printings included).
Here is a screenshot from the logs:
![Screenshot 2023-06-13 at 8 46 59](https://github.com/noobaa/noobaa-core/assets/57721533/6e215f6b-25b8-4afb-a761-a87c24358a6d)

Reply:
why do you need to preserve the order of the keys?
I meant that it'll be a lookup Set for the duplicate check.
see the accepted answer here - https://stackoverflow.com/questions/49507420/lodash-performance-hit-with-uniqwith-to-remove-duplicate-objects

Reply:
You can read about `uniqWith()` [here](https://stackoverflow.com/questions/31740155/lodash-remove-duplicates-from-array) if you want (more users mark it as a useful answer).
I improved the performance check according to the link you send by using `performance.now()`, and you can see:
<img width="742" alt="Screenshot 2023-06-18 at 14 11 54" src="https://github.com/noobaa/noobaa-core/assets/57721533/f9a5acfa-7a31-4947-83d4-553b2c9d8768">

It's about ~8 ms
![Screenshot 2023-06-18 at 11 14 47](https://github.com/noobaa/noobaa-core/assets/57721533/5db4b02c-9834-4dc4-ac57-cef5e72be5f7)

Reply:
`Set` is using the `===` operator (see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#value_equality)), hence we cannot compare the whole objects (since `===` comparison on objects checks reference comparison) and we would need another tool to check the two properties of `key` and `version_id` to decide if an object is duplicated.

Reply:
try the following method:
```
// method object 
const obj_start = performance.now();
const obj = {};
const obj_result = [];
for (let i = 0; i < objects.length; i++) {
  const key_version = JSON.stringify(objects[i]);
  if (!obj[key_version]) {
    obj[key_version] = true;
    obj_result.push(key_version);
  }
}
const obj_end = performance.now();
```
I see a major difference when running on my laptop between the 3 methods: set, object and uniq.
SET:
Removed 6 duplicates from 1000 items
Operation took 0.7413411140441895 ms
OBJECT:
Removed 6 duplicates from 1000 items
Operation took 1.0831308364868164 ms
UNIQUE:
Removed 6 duplicates from 1000 items
Operation took 378.04422998428345 ms

Reply:
if you will push objects[i]  to obj_result will it work without parsing it again?

Reply:
Yes